diff --git a/generate-ios-source-and-headers.py b/generate-ios-source-and-headers.py
index c2bca73..ec810d6 100644
--- a/generate-ios-source-and-headers.py
+++ b/generate-ios-source-and-headers.py
@@ -125,11 +125,12 @@ def build_target(platform):
         os.makedirs(build_dir)
         env = dict(CC=xcrun_cmd('clang'),
                    LD=xcrun_cmd('ld'),
-                   CFLAGS='-arch %s -isysroot %s -miphoneos-version-min=4.0' % (platform.arch, platform.sdkroot))
+                   CFLAGS='-arch %s -isysroot %s -miphoneos-version-min=4.0 -no-integrated-as' % (platform.arch, platform.sdkroot))
         working_dir=os.getcwd()
+        install_dir = os.path.join(working_dir, platform.arch + '-ios')
         try:
             os.chdir(build_dir)
-            subprocess.check_call(['../configure', '-host', platform.triple], env=env)
+            subprocess.check_call(['../configure', '-prefix', install_dir, '-host', platform.triple], env=env)
             move_source_tree('.', None, '../ios/include',
                              arch=platform.arch,
                              prefix=platform.prefix,
